### PR TITLE
[#2083] feat(core): Add JDBC backend operations for table

### PR DESCRIPTION
**Pull request by xloya**
### What changes were proposed in this pull request?

The purpose of this PR is to implement JDBC backend operation for Table metadata. Depend on #2377 . Metadata operations of Fileset will be supported in the remaining PRs.

### Why are the changes needed?

Fix: #2083 

### How was this patch tested?

Add unit tests to test the table metadata ops.

### CODE REVIEW

**Comment by xloya** [comment]
@yuqi1129 @jerryshao @qqqttt123 This PR is ready for reviews, please take a look when you have time, thanks!

File: core/src/main/resources/mysql/mysql_init.sql [hunk 1]
```diff
@@ -47,4 +47,20 @@ CREATE TABLE IF NOT EXISTS `schema_meta` (
...
+    `schema_id` BIGINT(20) UNSIGNED NOT NULL COMMENT 'schema id',
+    `audit_info` MEDIUMTEXT NOT NULL COMMENT 'table audit info',
+    `current_version` INT UNSIGNED NOT NULL DEFAULT 1 COMMENT 'table current version',
+    `last_version` INT UNSIGNED NOT NULL DEFAULT 1 COMMENT 'table last version',
+    `deleted_at` BIGINT(20) UNSIGNED NOT NULL DEFAULT 0 COMMENT 'table deleted at',
+    PRIMARY KEY (`table_id`),
+    UNIQUE KEY `uk_sid_tn_del` (`schema_id`, `table_name`, `deleted_at`),
+    KEY `idx_mid` (`metalake_id`),
+    KEY `idx_cid` (`catalog_id`)
+) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_bin COMMENT 'table metadata';
```
Comment (jerryshao) [inline core/src/main/resources/mysql/mysql_init.sql:66]: I remembered that for `table` metadata, you will have another "version" table, am I right or will you do it in another PR?

Comment (xloya) [reply core/src/main/resources/mysql/mysql_init.sql:66]: There is no need to create a `version` table for the properties that TableEntity currently contains, so I did not create a `version` table for `TableEntity` in this PR. If `TableEntity` needs to add more properties and maintain multiple versions in the future, we can create the `version` table separately. For `FilesetEntity` I will create a `version` table. This is the current properties of `TableEntity`:
image

File: core/src/main/java/com/datastrato/gravitino/storage/relational/service/TableMetaService.java [hunk 2]
```diff
@@ -0,0 +1,190 @@
...
+          re, Entity.EntityType.TABLE, newEntity.nameIdentifier().toString());
+      throw re;
+    }
+
+    if (updateResult > 0) {
+      return newEntity;
+    } else {
+      throw new IOException("Failed to update the entity: " + identifier);
+    }
+  }
```
Comment (jerryshao) [inline core/src/main/java/com/datastrato/gravitino/storage/relational/service/TableMetaService.java:148]: This update operation seems not transactional, but it's fine as we have locks in the outside.

File: core/src/main/java/com/datastrato/gravitino/storage/relational/utils/POConverters.java [hunk 3]
```diff
@@ -303,4 +305,88 @@ public static List<SchemaEntity> fromSchemaPOs(List<SchemaPO> schemaPOs, Namespa
...
+  public static TablePO initializeTablePOWithVersion(
+      TableEntity tableEntity, TablePO.Builder builder) {
+    try {
+      return builder
+          .withTableId(tableEntity.id())
+          .withTableName(tableEntity.name())
+          .withAuditInfo(JsonUtils.anyFieldMapper().writeValueAsString(tableEntity.auditInfo()))
+          .withCurrentVersion(1L)
+          .withLastVersion(1L)
+          .withDeletedAt(0L)
```
Comment (jerryshao) [inline core/src/main/java/com/datastrato/gravitino/storage/relational/utils/POConverters.java:324]: Can we not hardcode 0L here, instead we'd better define a constant to describe it.

File: core/src/main/java/com/datastrato/gravitino/storage/relational/service/TableMetaService.java (hunk 2 above)
Comment (xloya) [reply core/src/main/java/com/datastrato/gravitino/storage/relational/service/TableMetaService.java:148]: I learned that the current plan is to implement locks in the upper layer, because backends without transaction capabilities may be supported, so more complex locks are not implemented here.

File: core/src/main/java/com/datastrato/gravitino/storage/relational/utils/POConverters.java (hunk 3 above)
Comment (xloya) [reply core/src/main/java/com/datastrato/gravitino/storage/relational/utils/POConverters.java]: Done

**Review by jerryshao — APPROVED** [review]
Overall LGTM, @yuqi1129 @qqqttt123 please take time to review this.

**Review by qqqttt123 — APPROVED** [review]
